### ISSUE TITLE
Redis6Proxy incompatible with phpredis

### ISSUE THREAD

**Issue by 111ypuk** [symfony/symfony#57884]
### Symfony version(s) affected

7.1

### Description

After update https://github.com/phpredis/phpredis/commit/6ea5b3e08bdbf8cbe93e0dc56b18e8316d65097c

Get error:

```
 Declaration of Symfony\Component\Cache\Traits\Redis6Proxy::hSet($key, $member, $value): Redis|int|false must be compatible with Redis::hSet(string $key, mixed ...$fields_and_vals): Redis|int|false {"file":"/app/vendor/symfony/cache/Traits/Redis6Proxy.php","exception":"[object] (ErrorException(code: 0): Declaration of Symfony\\Component\\Cache\\Traits\\Redis6Proxy::hSet($key, $member, $value): Redis|int|false must be compatible with Redis::hSet(string $key, mixed ...$fields_and_vals): Redis|int|false at /app/vendor/symfony/cache/Traits/Redis6Proxy.php:519)"
````

### How to reproduce

Create a symfony web project and inside it a Dockerfile with a php container which installs redis
...
FROM php:8.3.8-cli-alpine
...
RUN apk add --no-cache lz4-dev \
    && mkdir /tmp/phpredis \
    && cd /tmp/phpredis \
    && git clone --recurse-submodules https://github.com/phpredis/phpredis "php-redis" \
    && cd php-redis \
    && phpize \
    && ./configure --enable-redis-igbinary --enable-redis-lz4 --with-liblz4=/usr/lib \
    && make \
    && make install \
    && make clean \
    && docker-php-ext-enable redis \
    && rm -rf /tmp/phpredis
...

Run the docker container and open the website

**Comment by xabbuh** [comment]
Thank you for the report. Can you please check if #57885 would solve the issue that you experience?

**Comment by 111ypuk** [comment]
> Thank you for the report. Can you please check if #57885 would solve the issue that you experience?

yes

**Comment by nuryagdym** [comment]
after the upgrade to symfony v6.4.12 (which contains this changes) I still had this issue.
running `bin/console cache:clear` continuing to give this error.

deleting /var/cache folder and then running `bin/console cache:clear` solved the issue.

**Comment by anselmos** [comment]
This is not working on 6.2 - is there any way to fix it besides upgrading to v6.4.12 ?

**Comment by stof** [comment]
No. 6.2 is not maintained anymore and so was not patched.
